developer/frameworks/articles/dependency-injection/02-cdi-dependency-injection.markdown

```diff
@@ -0,0 +1,142 @@
+---
+header-id: cdi-dependency-injection
+---
+
+# CDI Dependency Injection
+
+[TOC levels=1-4]
+
+Portlet 3.0 (see [JSR 362](https://jcp.org/en/jsr/detail?id=362)) supports using
+Contexts and Dependency Injection (CDI). It enables you to create and use
+injectable classes (CDI beans) in your portlet. It also provides injectable
+portlet artifacts called [Portlet Predefined
+Beans](/docs/7-2/reference/-/knowledge_base/r/cdi-portlet-predefined-beans).
+They give a portlet's CDI beans access to the portlet configuration,
+preferences, requests, responses, and more. Here you'll create and use CDI beans
+and use Portlet Predefined Beans in your portlet. 
+
+1.  Create a portlet WAR project, if you haven't created one 
+    already. 
+
+    -   [Bean Portlet](/docs/7-2/appdev/-/knowledge_base/a/bean-portlet) 
+    -   [JSF Portlet](/docs/7-2/appdev/-/knowledge_base/a/jsf-portlet) 
+    -   [PortletMVC4Spring
+        Portlet](/docs/7-2/appdev/-/knowledge_base/a/portlet-mvc-for-spring-portletmvc4spring) 
```

Comment:
@jhinkey @sez11a Recommend removing any mention of PortletMVC4Spring since this article is about CDI. When 7.2 GA2 comes out, it will contain support for using JSR 330 `@Inject` with either CDI or Spring. But that has nothing to do with PortletMVC4Spring which does not rely on the bean portlet feature.